app/models/business_entity.rb

```diff
@@ -2,5 +2,15 @@ class BusinessEntity < ActiveRecord::Base
   belongs_to :office
   belongs_to :jurisdiction
 
-  validates :office, :jurisdiction, :code, :name, presence: true
+  validates :office, :jurisdiction, :code, :name, :valid_from, presence: true
+
+  validates :valid_to, date: {
+    after: :valid_from, allow_blank: true
+  }
+
+  scope :current_by_office, lambda { |office_id|
```

Comment:
I'm thinking that this should either be a `default` scope, or for example the `Office#business_entities` should use this scope. I'm not sure what the right solution is.